helix-view/src/document.rs

```diff
@@ -733,16 +733,25 @@ impl Document {
     // We can't use anyhow::Result here since the output of the future has to be
     // clonable to be used as shared future. So use a custom error type.
     pub fn format(&self) -> Option<BoxFuture<'static, Result<Transaction, FormatterError>>> {
-        if let Some(formatter) = self
+        if let Some((fmt_cmd, fmt_args)) = self
             .language_config()
             .and_then(|c| c.formatter.clone())
-            .filter(|formatter| which::which(&formatter.command).is_ok())
+            .and_then(|formatter| {
+                Some((
+                    which::which(&formatter.command)
+                        .ok()?
+                        .into_os_string()
+                        .into_string()
+                        .ok()?,
```

Comment:
Ah one thing I missed here, paths are not necessarily utf-8 you should just return a path here instead of converting to string